### PR TITLE
DashboardPicker: switch to promise-based debounce, return dashboard UID

### PR DESCRIPTION
**Pull request by Clarity-89**
**What this PR does / why we need it**:
- Use [debounce-promise](https://github.com/bjoerge/debounce-promise) instead of lodash's debounce for `loadOptions` function. The promise-based approach is necessary because otherwise the results fo the first call will overwrite the results for the debounced call when filtering options. The issue is described in great detail [here](https://github.com/JedWatson/react-select/issues/3075#issuecomment-450194917). 
- Return dashboard UID from the search, so it can be used for retrieving dashboards via API.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -89,6 +89,7 @@
     "@types/d3": "5.7.2",
     "@types/d3-force": "^2.1.0",
     "@types/d3-scale-chromatic": "1.3.1",
+    "@types/debounce-promise": "^3.1.3",
```
Comment (hugohaggmark) [inline package.json:92]: nit: always use exact versions

File: package.json [hunk 2]
```diff
@@ -234,6 +235,7 @@
     "d3-force": "^2.1.1",
     "d3-scale-chromatic": "1.5.0",
     "dangerously-set-html-content": "1.0.6",
+    "debounce-promise": "^3.1.2",
```
Comment (hugohaggmark) [inline package.json:238]: nit: always use exact versions

File: package.json (hunk 1 above)
Comment (Clarity-89) [reply package.json:92]: True, great catch!

**Review by hugohaggmark — APPROVED** [review]
Looks great and works as expected 🎉

**Pull request merged**